assemble-and-restart

```diff
@@ -4,12 +4,11 @@ set -eo pipefail
 
 # We now run the assembly script. If there is a custom one written in the
 # source files, we use that instead.
-if [ -f /opt/app-root/src/.s2i/bin/assemble ]; then
-    /opt/app-root/src/.s2i/bin/assemble
-elif [ -f /usr/local/s2i/assemble ]; then # java s2i
-    rm -rf /opt/app-root/src/.git # ensure we don't copy git files since they can cause problems
-    cp -a /opt/app-root/src/. /tmp/src # java s2i expects the source to be under /tmp/src
-    /usr/local/s2i/assemble
+if [ -f ${ODO_S2I_SRC_BIN_PATH}/.s2i/bin/assemble ]; then
+    ${ODO_S2I_SRC_BIN_PATH}/.s2i/bin/assemble
+elif [ -n "${ODO_S2I_SCRIPTS_URL}" ]; then # For S2I scripts path, use the env var set by odo if not available in component source
+    rm -rf ${ODO_S2I_SRC_BIN_PATH}/.git # ensure we don't copy git files since they can cause problems
```

Comment:
Also after this statement, we still need to copy artifacts into `${ODO_S2I_SRC_BIN_PATH}/src` using `cp -a /opt/app-root/src/. ${ODO_S2I_SRC_BIN_PATH}/src` if and only if `ODO_S2I_SRC_BIN_PATH != /opt/app-root`

Reply:
A way to do things correctly would be:

```
elif [ -n "${ODO_S2I_SCRIPTS_URL}" ]; then # For S2I scripts path, use the env var set by odo if not available in component source
    if [ "${ODO_S2I_SRC_BIN_PATH}" != "/opt/app-root/src" ]; then
      rm -rf ${ODO_S2I_SRC_BIN_PATH}/.git # ensure we don't copy git files since they can cause problems
      cp -a /opt/app-root/src/. ${ODO_S2I_SRC_BIN_PATH}
    fi
    ${ODO_S2I_SCRIPTS_URL}/assemble
```